charts/datadog/templates/system-probe-configmap.yaml

```diff
@@ -21,10 +32,12 @@ data:
       enable_tcp_queue_length: {{ $.Values.datadog.systemProbe.enableTCPQueueLength }}
       enable_oom_kill: {{ $.Values.datadog.systemProbe.enableOOMKill }}
       collect_dns_stats: {{ $.Values.datadog.systemProbe.collectDNSStats }}
-    {{- if $.Values.datadog.networkMonitoring.enabled }}
     network_config:
-      enabled: true
-    {{- end }}
+      {{- if (eq ($.Values.datadog.networkMonitoring.enabled | toString) "<nil>") }}
```

Comment:
`datadog.networkMonitoring.enabled` from the `values.yaml` file should be `false` or `true`, so why checking nil ?

Reply:
Long story short, we need that sentinel value for backwards compatibility. We want to meet the following conditions:

* systemProbe.enabled -> defaults to false
* networkConfig.enabled -> defaults to false
* systemProbe.enabled= false, networkConfig.enabled = not set -> system probe is off
* systemProbe.enabled=false, networkConfig.enabled = false -> system probe is off
* systemProbe.enabled=false, networkConfig.enabled=true -> system probe is on, but only the network module
* systemProbe.enabled = true, networkConfig.enabled = not set -> system probe is on, with network module enabled (for back compat reasons)
* systemProbe.enabled = true, networkConfig.enabled = false (explicitly) -> system probe is on, with network module disabled
* systemProbe.enabled = true, networkConfig.enabled = true (explicitly) -> system probe is on, with network module enabled

Reply:
The issue is that the auto-generated docs now want this variable to be documented as `string` which might be misleading to customers :(
```
+| datadog.networkMonitoring.enabled | string | `nil` | Enable network performance monitoring |
```

Reply:
nevermind, found a workaround

Reply:
you need also to consider the `securityagent.runtime` and `[oom,tcpqueue] probe` in systemprobe enabling process.

Reply:
I'm not sure what you mean by your comment. System-probe enabling is already being enabled by these flags.

Reply:
I think having this "nil/null" value will confuse users.

In previous a PR, we deprecate "systemProbe.enabled" option. The idea was to not consider the `systemProbe` as a feature, but just an implementation details.

And so instead of a global `systemProbe.enabled`, we have for each feature/product a boolean to enabled the corresponding module in `systemProbe`. 

From your previous use cases description, I think the only case which is problematic is `systemProbe.enable=true` with `networkMonitoring.enabled=false`.
But we can avoid having this case by adding a test in the chart, that refuses the deployment, and return an error message that said: "networkMonitoring can't be disable if `systemProbe.enable=true`, please remove the "systemProbe.enable" option from your configuration."

WDYT?

Reply:
That sounds reasonable to me. @leeavital any objections to this?

Reply:
Are you suggesting that:
* the defaults are `networkMonitoring.enabled=false` && `systemProbe.enabled=false`   
* `networkMonitoring.enabled=false` && `systemProbe.enabled=true` causes the chart to fail
?

I think that's fine for new users of the chart, but breaks all the customers who have been using the old form of config.  Generally, as a helm user, I expect upgrading a chart to "just work" and this will break that.

Perhaps I'm misunderstanding what you're proposing ?

Reply:
We advice users since several versions to avoid using systemProbe.enabled. 
So I think it ok to move forward and try to enforce proper configuration. Do make this change we can increment the minor version.

Reply:
It has been only a few months since we introduced the `networkMonitoring` field.     I don't have a good sense of how long is "right".    

If we already have precedent for breaking configurations (and it sounds like we do) to enforce config migrations, then I think your suggestion makes a lot of sense.

Reply:
@clamoriniere I fully removed `systemProbe.enabled` option and did a minor version bump. Please give it a review let us know if that makes sense.